### DISCUSSION TITLE
possible to remove VLAN?

### DISCUSSION THREAD

**Discussion by spencersevilla** [srsran/srsRAN_Project#626] [General Help] [answered]
Hi everyone,

It looks like no matter what I do, the DU adds a VLAN tag to the packets it sends to the RU. If I comment out `vlan_tag_cp` (or `_up`) it uses a default value of 1, and if I set `vlan_tag_cp = 0` it throws an error (since it's an invalid tag ID). I have the RU plugged directly into the DU and they are not set up with any sort of VLAN, so I would like to remove the VLAN tag from the transmitted packets. Before I start hacking around on this I was just wondering if I was missing something here? Feels like a strange architectural choice to enforce. Also soliciting feedback on whether the community would appreciate this as a feature-add and the best way to write the code if so. I figured I would leave the default behavior unchanged and use `vlan_tag_cp = 0` to indicate that no VLAN is to be used.

**Comment by ffxxcad** [comment]
Other users like [this one](https://github.com/srsran/srsRAN_Project/discussions/398#discussion-6055892) connect directly to the RU as well. I can't speak for SRS but I believe that the VLAN tag parameter exists because RUs like Benetel require tagged packets.

From what I've seen, as long as the VLAN tag in your DU config matches what you have in the RU configuration it should work.

**Reply by spencersevilla** [reply]
> From what I've seen, as long as the VLAN tag in your DU config matches what you have in the RU configuration it should work.

Thats exactly the problem, my RU isn't setup with a VLAN at all! Thanks for the context about Benetel requiring tagged packets, that definitely gives me some useful insight as to why it's setup like this.

**Reply by andrepuschmann** [reply, marked as answer]
It's very uncommon to have an RU without VLAN for CU plane. S plane is untagged by the spec but CU plane is usually tagged. Currently there is no config option to remove the VLAN header from the OFH packets. But you can configure your switch to untag the port you've connected the RU on.

**Reply by spencersevilla** [reply]
Just wanted to drop a note and say thanks for the context/explanation! i'm still pretty new to ORAN and didn't know VLAN was so standard; I'll solve this issue the way you describe.
